### ISSUE TITLE
Difficulty in achieving similar improvements in FIQA for few-shot learning as reported in table 3

### ISSUE THREAD

**Issue by xhluca** [facebookresearch/contriever#16]
I found Contriever quite interesting based on the table 3 of the paper (few-shot retrieval) as Contriever-MSMarco achieves a score of 38.1 when finetuned on FiQA, which is much higher than the BERT which is at 30. 

I’ve tried a similar set up, with the differences being:
1.	Trained for 20 epochs instead of 500
2.	AdamW instead of ASAM
3.	Included BM25 hard negatives (i.e. top results that are not a gold label) in addition to in-batch negative sampling
4.	Batch size of 128 instead of 256 (though the number of negatives should be the same due to HNs)
5.     Instead of early stopping, I just trained for 20 epochs and selected the best model.

It seems that under those settings, the improvements isn't as high as the difference reported in the paper:

| split   |   epoch | metric   | model_name                  |   learning_rate |     k=1 |     k=3 |     k=5 |    k=10 |   k=100 |   k=1000 |
|:--------|--------:|:---------|:----------------------------|----------------:|--------:|--------:|--------:|--------:|--------:|---------:|
| test    |       7 | ndcg     | facebook/contriever-msmarco |           1e-05 | 0.24383 | 0.25005 | 0.2608  | 0.28715 | 0.36118 |  0.39975 |
| test    |       3 | ndcg     | facebook/contriever-msmarco |           3e-05 | 0.26698 | 0.25233 | 0.25753 | 0.28208 | 0.35664 |  0.39693 |
| test    |      16 | ndcg     | facebook/contriever         |           3e-05 | 0.25    | 0.23583 | 0.24952 | 0.2732  | 0.35149 |  0.39019 |
| test    |       9 | ndcg     | facebook/contriever         |           1e-05 | 0.24537 | 0.23263 | 0.24548 | 0.2702  | 0.35148 |  0.3906  |
| test    |       6 | ndcg     | facebook/contriever-msmarco |           5e-05 | 0.24383 | 0.23366 | 0.24735 | 0.2699  | 0.34569 |  0.38747 |
| test    |       2 | ndcg     | facebook/contriever         |           5e-05 | 0.23457 | 0.22305 | 0.24112 | 0.26565 | 0.34186 |  0.38429 |
| test    |      12 | ndcg     | roberta-base                |           5e-05 | 0.25309 | 0.22701 | 0.24416 | 0.26293 | 0.33809 |  0.37927 |
| test    |      13 | ndcg     | roberta-base                |           4e-05 | 0.22685 | 0.21018 | 0.22893 | 0.25435 | 0.33068 |  0.3722  |
| test    |      15 | ndcg     | roberta-base                |           3e-05 | 0.21605 | 0.20675 | 0.22277 | 0.25412 | 0.32839 |  0.36796 |
| test    |      19 | ndcg     | roberta-base                |           2e-05 | 0.23148 | 0.21632 | 0.22439 | 0.25082 | 0.3299  |  0.37125 |
| test    |      18 | ndcg     | roberta-base                |           1e-05 | 0.19599 | 0.19298 | 0.20674 | 0.23406 | 0.31369 |  0.35424 |

Note the NDCG@10 of the best contriever model is 1.5 higher than the best roberta-base (I did not try BERT). Is there something that I may be missing? It seems odd that Roberta-base achieved results similar to what was reported in the paper (26.1 for BERT vs 26.3 in my 
case) but not for contriever.

I'd be happy to try running a huggingface checkpoint of your model trained on Fiqa, but I was unable to find that. I did not find scripts that lets me reproduce the results on FiQA in this repository.

Thank you!

**Comment by siyuanseever** [comment]
I have the same requirements and encountered similar problems, my few-shot implementation scores are much worse than those in the paper.
